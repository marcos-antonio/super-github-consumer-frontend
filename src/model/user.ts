export class User {
  constructor(user: Partial<User>) {
    this.id = user.id;
    this.login = user.login;
    this.htmlUrl = user.htmlUrl;
    this.avatarUrl = user.avatarUrl;
    this.createdAt = user.createdAt;
  }

  id: number;
  login: string;
  htmlUrl: string;
  avatarUrl: string;
  createdAt: string;
}
