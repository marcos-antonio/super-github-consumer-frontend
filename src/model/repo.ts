import { User } from './user';

export class Repo {
  constructor(repo: Partial<Repo>) {
    this.id = repo.id;
    this.name = repo.name;
    this.htmlUrl = repo.htmlUrl;
    this.owner = repo.owner;
  }

  id: number;
  name: string;
  htmlUrl: string;

  owner: User;
}
