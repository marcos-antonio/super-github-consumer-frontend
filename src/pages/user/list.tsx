import React from 'react'

import { UserListTemplate } from 'templates/user/list'

const Index = () => <UserListTemplate />

export default Index
