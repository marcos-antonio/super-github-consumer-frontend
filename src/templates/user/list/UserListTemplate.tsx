import React, { FunctionComponent } from 'react';
import { useRouter } from 'next/router';

import { Table, HeaderCell, Row } from 'components/table';
import { User } from 'model/user';

import * as S from './styled';

export interface UserListTemplateProps {
  userList: User[];
  hasPagination?: boolean;
  nextPageClickHandler?: () => void;
  previousPageClickHandler?: () => void;
  previousPageDisabled?: boolean;
  nextPageDisabled?: boolean;
}

export const UserListTemplate: FunctionComponent<UserListTemplateProps> = ({
  userList,
  hasPagination,
  nextPageClickHandler,
  previousPageClickHandler,
  nextPageDisabled = false,
  previousPageDisabled = false,
}) => {
  const header: HeaderCell[] = [{ child: 'ID' }, { child: 'Login' }];

  const router = useRouter();
  const rows: Row[] = userList.map(data => ({
    columns: [data.id, data.login],
  }));
  const detailUser = (userLogin: string) => {
    router.push(`/users/${userLogin}`);
  };

  const paginationProps = {
    nextPageClickHandler,
    previousPageClickHandler,
    nextPageDisabled,
    previousPageDisabled,
  };

  return (
    <S.Container>
      <S.TableContainer data-testid="table-container">
        <Table
          header={header}
          rows={rows}
          rowClickHandler={row => {
            const userLogin = row.columns[1].toString();
            if (userLogin) detailUser(userLogin);
          }}
          paginationProps={hasPagination && paginationProps}
        />
      </S.TableContainer>
    </S.Container>
  );
};
