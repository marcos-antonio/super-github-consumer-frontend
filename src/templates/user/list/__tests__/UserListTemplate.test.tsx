import React from 'react';
import { render, RenderResult } from '@testing-library/react';

import { ThemeProvider } from 'styled-components';

import theme from 'components/Theme';
import { User } from 'model/user';

import { UserListTemplate } from '../UserListTemplate';

describe('Component: UserListTemplate', () => {
  let wrapper: RenderResult;

  const userList = [
    new User({
      id: 2,
      login: 'defunkt',
      htmlUrl: 'https://github.com/defunkt',
      createdAt: '2011-01-25T18:44:36Z',
    }),
  ];

  beforeEach(() => {
    wrapper = render(
      <ThemeProvider theme={theme}>
        <UserListTemplate userList={userList} />
      </ThemeProvider>
    );
  });

  it('render the component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
