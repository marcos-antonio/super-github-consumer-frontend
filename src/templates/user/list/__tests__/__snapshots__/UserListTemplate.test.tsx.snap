// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: UserListTemplate render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="styled__Container-sc-1ozttvw-0 jmiUhe"
      >
        <div
          class="styled__TableContainer-sc-1ozttvw-1 hrtKVn"
          data-testid="table-container"
        >
          <div
            class="styled__Container-q0w278-0 iFBbUj"
          >
            <div
              class="styled__Table-q0w278-1 eJXAqz"
            >
              <div
                class="styled__HeaderContainer-q0w278-2 fUNRQo"
              >
                <div
                  class="styled__Header-q0w278-3 klUxVQ"
                >
                  <div
                    class="styled__HeaderCell-q0w278-4 cXSibf"
                  >
                    ID
                  </div>
                  <div
                    class="styled__HeaderCell-q0w278-4 cXSibf"
                  >
                    Login
                  </div>
                </div>
              </div>
              <div
                class="styled__Body-q0w278-5 gHKXfy"
              >
                <div
                  class="styled__RowContainer-q0w278-6 gXiuGM"
                >
                  <div
                    class="styled__Row-q0w278-7 htLHt"
                  >
                    <div
                      class="styled__RowCell-q0w278-8 fTlsqw"
                    >
                      2
                    </div>
                    <div
                      class="styled__RowCell-q0w278-8 fTlsqw"
                    >
                      defunkt
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="styled__PaginationContainer-q0w278-9 huNEmJ"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="styled__Container-sc-1ozttvw-0 jmiUhe"
    >
      <div
        class="styled__TableContainer-sc-1ozttvw-1 hrtKVn"
        data-testid="table-container"
      >
        <div
          class="styled__Container-q0w278-0 iFBbUj"
        >
          <div
            class="styled__Table-q0w278-1 eJXAqz"
          >
            <div
              class="styled__HeaderContainer-q0w278-2 fUNRQo"
            >
              <div
                class="styled__Header-q0w278-3 klUxVQ"
              >
                <div
                  class="styled__HeaderCell-q0w278-4 cXSibf"
                >
                  ID
                </div>
                <div
                  class="styled__HeaderCell-q0w278-4 cXSibf"
                >
                  Login
                </div>
              </div>
            </div>
            <div
              class="styled__Body-q0w278-5 gHKXfy"
            >
              <div
                class="styled__RowContainer-q0w278-6 gXiuGM"
              >
                <div
                  class="styled__Row-q0w278-7 htLHt"
                >
                  <div
                    class="styled__RowCell-q0w278-8 fTlsqw"
                  >
                    2
                  </div>
                  <div
                    class="styled__RowCell-q0w278-8 fTlsqw"
                  >
                    defunkt
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="styled__PaginationContainer-q0w278-9 huNEmJ"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
